### PR TITLE
Changeling Biodegrade Power, Fixes Hivemind Link

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Biodegrade costs 2 evolution points and uses up 30 chems. It lets you spit acid on handcuffs/straightjacket/locker/cocoon to destroy them and escape.

Hivemind Link was added to the code and worked fine but someone accidentally removed it from the .dme some time ago so it wasn't appearing in-game. Readded it to the .dme.

Moved some closet/crate/locker vars up to closets.dm

from /tg/

:cl:
add: Changelings have a new power, Biodegrade, for 2 evolution points. For 30 chemicals you can vomit acid onto restraints, welded/locked lockers and spider cocoons to escape them.
fix: Hivemind Link for changelings has been readded, it was removed in error
/:cl:

### CODE REVIEW

File: code/game/gamemodes/changeling/powers/biodegrade.dm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	desc = "Dissolves restraints or other objects preventing free movement."
+	helptext = "This is obvious to nearby people, and can destroy standard restraints and closets."
+	chemical_cost = 30 //High cost to prevent spam
+	dna_cost = 2
+	req_human = 1
+
+/obj/effect/proc_holder/changeling/biodegrade/sting_action(mob/living/carbon/human/user)
+	var/used = FALSE // only one form of shackles removed per use
+	if(!user.restrained() && !istype(user.loc, /obj/structure/closet) && !istype(user.loc, /obj/structure/spider/cocoon))
+		to_chat(user, "<span class='warning'>We are already free!</span>")
```
Comment (taukausanake) [inline code/game/gamemodes/changeling/powers/biodegrade.dm]: "We"? I wasn't aware changelings talk in the, what is that third person? Besides that I thought all those notifications were just that like "You are already free"

Comment (Birdtalon) [reply code/game/gamemodes/changeling/powers/biodegrade.dm]: We is used on other changeling spells also.

Comment (taukausanake) [reply code/game/gamemodes/changeling/powers/biodegrade.dm]: That's weird. Changelings are still one being, just with multiple faces. I guess they do get some of the previous memories of the person they suck. I must be missing something about changelings

**Comment by KingPhilipIII** [comment]
Changelings not being defeated by decapitation when. I want to be the headless horseless horseman.

**Comment by taukausanake** [comment]
@KingPhilipIII What if wizard horse curse glitches with your headless changeling? That would be _amazing_

File: code/game/gamemodes/changeling/powers/biodegrade.dm (hunk 1 above)
Comment (FalseIncarnate) [reply code/game/gamemodes/changeling/powers/biodegrade.dm]: Hivemind, meaning a single changeling is still part of a collective, hence the use of the inclusive collective pronouns. "We must absorb Baldie McGriffin", "We reconfigure ourselves", etc

**Comment by KingPhilipIII** [comment]
@taukausanake  I guess I would technically be the horseless horseman then? I would have ahead, but I'd be a horseman. Literally.

File: code/game/gamemodes/changeling/powers/biodegrade.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	if(used)
+		feedback_add_details("changeling_powers","BD")
+	return TRUE
+
+/obj/effect/proc_holder/changeling/biodegrade/proc/dissolve_handcuffs(mob/living/carbon/human/user, obj/O)
+	if(O && user.handcuffed == O)
+		user.unEquip(O)
+		O.visible_message("<span class='warning'>[O] dissolves into a puddle of sizzling goop.</span>")
+		O.loc = get_turf(user)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/biodegrade.dm]: `O.forceMove(get_turf(user))`

File: code/game/gamemodes/changeling/powers/biodegrade.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		qdel(O)
+
+/obj/effect/proc_holder/changeling/biodegrade/proc/dissolve_straightjacket(mob/living/carbon/human/user, obj/S)
+	if(S && user.wear_suit == S)
+		user.unEquip(S)
+		S.visible_message("<span class='warning'>[S] dissolves into a puddle of sizzling goop.</span>")
+		S.loc = get_turf(user)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/biodegrade.dm]: `S.forceMove(get_turf(user))`

**Comment by Fox-McCloud** [comment]
conflicts.

Also just a few small issues.

**Pull request merged**